lib/protobuf/nats/client.rb

```diff
@@ -65,6 +71,12 @@ def send_request
         # Nats response timeout.
         retry if (retries -= 1) > 0
         raise
+      rescue NackError
```

Comment:
as I was thinking last night I believe the "ideal" would be to make this throw/catch instead of rescue based; we know that we are in a "fail/retry" state and don't need to rely on an exception; also in Jruby the generation of an exception is slow/heavy

That's right ... 100x difference with a cold throw/catch from a warm rescue

```ruby
jruby-9.1.7.0 :002 > def res
jruby-9.1.7.0 :003?>   fail "derp"
jruby-9.1.7.0 :004?>   rescue => e
jruby-9.1.7.0 :005?>   nil
jruby-9.1.7.0 :006?>   end
 => :res 
jruby-9.1.7.0 :007 > def tc
jruby-9.1.7.0 :008?>   catch :something do
jruby-9.1.7.0 :009 >       end
jruby-9.1.7.0 :010?>   end
 => :tc 
jruby-9.1.7.0 :011 > def tc
jruby-9.1.7.0 :012?>   caught = false
jruby-9.1.7.0 :013?>   catch :something do
jruby-9.1.7.0 :014 >       return if caught
jruby-9.1.7.0 :015?>     caught = true
jruby-9.1.7.0 :016?>     throw :something
jruby-9.1.7.0 :017?>     end
jruby-9.1.7.0 :018?>   end
 => :tc 
jruby-9.1.7.0 :019 > Benchmark.measure do
jruby-9.1.7.0 :020 >     100_000.times do
jruby-9.1.7.0 :021 >       res
jruby-9.1.7.0 :022?>     end
jruby-9.1.7.0 :023?>   end
 => #<Benchmark::Tms:0x383bfa16 @stime=0.04999999999999999, @label="", @cstime=0.0, @real=13.571504258085042, @total=14.210000000000003, @cutime=0.0, @utime=14.160000000000002> 
jruby-9.1.7.0 :024 > Benchmark.measure do
jruby-9.1.7.0 :025 >     100_000.times do
jruby-9.1.7.0 :026 >       res
jruby-9.1.7.0 :027?>     end
jruby-9.1.7.0 :028?>   end
 => #<Benchmark::Tms:0x4229bb3f @stime=0.010000000000000009, @label="", @cstime=0.0, @real=13.664115638006479, @total=13.870000000000003, @cutime=0.0, @utime=13.860000000000003> 
jruby-9.1.7.0 :029 > Benchmark.measure do
jruby-9.1.7.0 :030 >     100_000.times do
jruby-9.1.7.0 :031 >       tc
jruby-9.1.7.0 :032?>     end
jruby-9.1.7.0 :033?>   end
 => #<Benchmark::Tms:0x543588e6 @stime=0.0, @label="", @cstime=0.0, @real=0.13869739905931056, @total=0.35999999999999943, @cutime=0.0, @utime=0.35999999999999943> 
```